### ISSUE TITLE
[Bug]: try/catch around the managerEntries causes static build to break

### ISSUE THREAD

**Issue by DustinJSilk** [storybookjs/storybook#20307]
### Describe the bug

After updating from `7.0.0-beta.8` to `7.0.0-beta.10`, running a static build with `storybook build -s public` causes an error in the browser: `unexpected token 'export'` in `storybook-static/sb-addons/manager.mjs`.

See this PR for the change that I believe caused the issue: https://github.com/storybookjs/storybook/commit/30119a6bec1d6c95b89b40af0d41b9efaaeffa1c

### To Reproduce

```shell
Build a new project
```

### System

```shell
System:
    OS: macOS 12.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 7.13.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 108.0.5359.124
    Firefox: 107.0
    Safari: 15.1
  npmPackages:
    @storybook/addon-a11y: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/addon-actions: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/addon-essentials: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/addon-interactions: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/addon-links: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/builder-vite: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/cli: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/html: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/html-vite: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/node-logger: 7.0.0-beta.10 => 7.0.0-beta.10 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1 
    @storybook/theming: 7.0.0-beta.10 => 7.0.0-beta.10
```

**Comment by aboutsimon** [comment]
This is fixed for me in version [v7.0.0-beta.19](https://github.com/storybookjs/storybook/releases/tag/v7.0.0-beta.19) (haven't tried an earlier release), but i guess it should be working since [v7.0.0-beta.15](https://github.com/storybookjs/storybook/releases/tag/v7.0.0-beta.15).

**Comment by shilman** [comment]
Thanks @aboutsimon -- closing!
